ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/resources/arjunacore/XAResourceRecord.java

```diff
@@ -678,6 +694,9 @@ public int topLevelOnePhaseCommit()
 	                    _theXAResource.commit(_tranID, true);
 	                else {
 	                    _theXAResource.rollback(_tranID);
+						// _rolledBack set to true even though
+						// it's not needed at this stage
+						this._rolledBack = true;
```

Comment:
Why would you do this? Would it not be set on line 309 or 346 above? Or is there another path?

Reply:
I'm not sure I understand this comment. Why would topLevelOnePhaseCommit depend on topLevelAbort?

Reply:
I will review!